### PR TITLE
RN-933 Limit file size in Meditrak file upload

### PR DESCRIPTION
**Pull request by IgorNadj**
### Changes:

- Never store file in database
- Set limit at 20 MB
- Make sure files are cleaned up

---

### Screenshots: 
https://linear.app/bes/issue/RN-933#comment-2b6d64f2

### CODE REVIEW

File: packages/meditrak-app/app/assessment/Question.js [hunk 1]
```diff
@@ -93,7 +93,7 @@ const mergeProps = ({ hasValidationErrorMessage, ...restOfStateProps }, { dispat
         dispatch(validateComponent(screenIndex, componentIndex, validationCriteria, newAnswer));
       }
     },
-    onChangeExtraProps: newProps => dispatch(changeExtraProps(id, newProps)),
```
Comment (rohan-bes) [inline packages/meditrak-app/app/assessment/Question.js:96]: I like this cleanup of `extraProps`. Can we take it one step further and just get rid of this whole thing? It seems like it was only used in the `File` and `Photo` questions

**Review by rohan-bes — APPROVED** [review]
Looks great! A few minor changes to request, mainly just the extraProps cleanup

**Review by rohan-bes — CHANGES_REQUESTED** [review]
Switching to request changes review cos I realised I wanted to take a second look after changes

File: packages/meditrak-app/app/sync/ChangeQueue.js [hunk 2]
```diff
@@ -48,19 +48,26 @@ export class ChangeQueue {
...
         break;
       }
       size += currentRecordSize;
       changesWithinThreshold.push({ change, payload: record });
+      const { action } = change;
+      if (action === 'AddSurveyFile') {
+        // If our files are quite large, and we load another one to see if it crosses our threshold,
+        // the Android app might run out of memory and crash. The workaround is to only ever
+        // upload one file at a time.
+        return changesWithinThreshold;
```
[the remark this answers is not in the document]
Comment (IgorNadj) [reply packages/meditrak-app/app/sync/ChangeQueue.js]: Missed this, fixed now, thanks!

File: packages/meditrak-app/app/assessment/specificQuestions/FileQuestion.js [hunk 3]
```diff
@@ -77,30 +118,48 @@ export const FileQuestion = ({ answer, extraProps, onChangeAnswer }) => {
...
+        return;
+      }
+
       onChangeAnswer(fileCopyUri);
+      setSizeInBytes(newSizeInBytes);
     } catch (e) {
+      await unlink(answer);
       onChangeAnswer(null);
+      setSizeInBytes(null);
+      setError(null);
```
[the remark this answers is not in the document]
Comment (IgorNadj) [reply packages/meditrak-app/app/assessment/specificQuestions/FileQuestion.js]: Added

File: packages/meditrak-app/app/assessment/Question.js (hunk 1 above)
Comment (IgorNadj) [reply packages/meditrak-app/app/assessment/Question.js:96]: Good point, the cleanup wasn't too bad

File: packages/meditrak-app/app/sync/Synchroniser.js [hunk 4]
```diff
@@ -403,4 +406,15 @@ export class Synchroniser {
...
     }
   }
+
+  async cleanupChangesAfterPush(changes) {
+    for (const change of changes) {
+      try {
+        await change.cleanupAfterPush(this.database);
+      } catch (e) {
+        // TODO: log
+        console.error(`Error thrown in cleanupAfterPush: ${e}`);
```
Comment (rohan-bes) [inline packages/meditrak-app/app/sync/Synchroniser.js:416]: I've noticed a few of these console.log calls around. I'm assuming that they're being left in intentionally?

**Review by rohan-bes — APPROVED** [review]
Nice stuff!

File: packages/meditrak-app/app/sync/Synchroniser.js (hunk 4 above)
Comment (IgorNadj) [reply packages/meditrak-app/app/sync/Synchroniser.js:416]: These ones are, yeah. I'm not sure how to hook into bugsnag other than crashing the whole app so I just left these as console.logs with a TODO. Seems like the TODO comment could be clearer?

**Pull request merged**